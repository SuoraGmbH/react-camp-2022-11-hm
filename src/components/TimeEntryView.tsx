import { TimeEntry } from "../domain/TimeEntry";

interface Props {
  timeEntry: TimeEntry;
}

const TimeEntryView = ({timeEntry}: Props) => {
  return (
    <div>
      <b>{timeEntry.comment.toUpperCase()}</b>
      <p>{timeEntry.start.toLocaleTimeString()}</p>
    </div>
  );
};

export default TimeEntryView;
