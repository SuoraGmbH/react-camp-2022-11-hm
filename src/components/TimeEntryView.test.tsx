import TimeEntryView from "./TimeEntryView";
import { render, screen } from "@testing-library/react";
import { TimeEntry } from "../domain/TimeEntry";

jest.useFakeTimers().setSystemTime(new Date("2022-08-25 13:42:23"))

const timeEntry: TimeEntry = {
  id: "adsoijijodas",
  comment: "Hallo hmmh!",
  start: new Date(),
  end: new Date(),
}

test("SmokeTest", () => {
  render(<TimeEntryView timeEntry={timeEntry} />)
});

test("it should display the comment", () => {
  render(<TimeEntryView timeEntry={timeEntry} />)
  // const commentElement = screen.queryByText("Hallo hmmh!")
  const commentElement = screen.queryByText(/hallo hmmh/i)
  expect(commentElement).toBeVisible()
});

test("it renders according to snapshot", () => {
  const {container} = render(<TimeEntryView timeEntry={timeEntry} />)
  expect(container).toMatchSnapshot()
})
