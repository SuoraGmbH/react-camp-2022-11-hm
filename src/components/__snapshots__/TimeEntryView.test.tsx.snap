// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders according to snapshot 1`] = `
<div>
  <div>
    <b>
      HALLO HMMH!
    </b>
    <p>
      1:42:23 PM
    </p>
  </div>
</div>
`;
